### PR TITLE
feat: Batching on `submitProof(s)` [CHAIN-407]

### PR DESCRIPTION
**Pull request by InversionSpaces**

### CODE REVIEW

File: test/core/capacity/CapacityCommitment.t.sol [hunk 1]
```diff
@@ -222,19 +225,118 @@ contract CapacityCommitmentTest is TestWithDeployment {
         vm.startPrank(peerOwner);
 
         bytes32 localUnitNonce = keccak256(abi.encodePacked("localUnitNonce"));
-        bytes32 targetHash = bytes32(uint256(deployment.core.difficulty()) - 1);
+        // RandomXProxyMock returns difficulty
+        bytes32 targetHash = deployment.core.difficulty();
 
         vm.expectEmit(true, true, true, false, address(deployment.capacity));
```
Comment (crystalbit) [inline test/core/capacity/CapacityCommitment.t.sol:231]: true, true, false, true

File: src/core/modules/capacity/Capacity.sol [hunk 2]
```diff
@@ -263,105 +263,145 @@ contract Capacity is UUPSUpgradeable, MulticallUpgradeable, BaseModule, ICapacit
...
+        }
+        // #endregion
 
-        bytes32 commitmentId = peer.commitmentId;
-        require(commitmentId != bytes32(0x00), "Compute unit doesn't have commitment");
+        bytes32[] memory globalUnitNonces = new bytes32[](unitIds.length);
 
-        Commitment storage cc = s.commitments[commitmentId];
-        require(currentEpoch >= cc.info.startEpoch, "Capacity commitment is not started");
+        for (uint256 i = 0; i < unitIds.length; i++) {
```
Comment (crystalbit) [inline src/core/modules/capacity/Capacity.sol:291]: pls length to separatee var (uint256 unitIdsLength = unitIds.length; for ...)

File: src/core/modules/capacity/Capacity.sol [hunk 3]
```diff
@@ -263,105 +263,145 @@ contract Capacity is UUPSUpgradeable, MulticallUpgradeable, BaseModule, ICapacit
...
         require(success, "RandomXProxy.run failed");
-        require(randomXResultHash.toBytes32() == resultHash, "Proof is not valid");
-        require(resultHash <= core.difficulty(), "Proof is bigger than difficulty");
+        require(result.length > 0, "RandomXProxy.run returned empty result");
+
+        bytes32[] memory hashes = abi.decode(result, (bytes32[]));
+
+        require(hashes.length == unitIds.length, "Invalid result length");
+
+        for (uint256 i = 0; i < unitIds.length; i++) {
```
Comment (crystalbit) [inline src/core/modules/capacity/Capacity.sol:388]: pls length to separatee var (uint256 unitIdsLength = unitIds.length; for ...)

**Review by crystalbit — APPROVED** [review]

File: src/utils/RandomXProxy.sol [hunk 4]
```diff
@@ -15,40 +15,79 @@ contract RandomXProxy {
 
     CommonTypes.FilActorId internal constant ActorID = CommonTypes.FilActorId.wrap(0x70768565);
```
Comment (folex) [inline src/utils/RandomXProxy.sol:16]: Should this be changed to `0x70768562`?

> define_id!(FLUENCE_BATCHED { id: 0x70768562 }); // ASCII of FLUB

**Pull request merged**